frappe/automation/doctype/assignment_rule/assignment_rule.py

```diff
@@ -37,6 +40,12 @@ def apply_unassign(self, doc, assignments):
 
 		return False
 
+	def validate_assign_conditions(self):
+		for field in ['assign_condition', 'unassign_condition', 'close_condition']:
+			condition = self.get(field, None)
+			if condition and ("=" in condition) and \
+				re.match(r"""[\w\.:_]+\s*={1}\s*[\w\.@'"]+""", condition):
+				frappe.throw(_("Invalid {0} condition").format(frappe.unscrub(field)), frappe.ValidationError)
```

Comment:
It's supposed to be simple python conditions right? Why not use `ast` or `safe_eval` to parse it?